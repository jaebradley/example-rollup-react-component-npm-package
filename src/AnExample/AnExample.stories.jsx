import React from 'react';

import AnExample from './index.jsx';

export default {
  title: 'AnExample',
  component: AnExample,
};

const Template = (args) => <AnExample {...args} />;

export const BasicExample = Template.bind({});
export const ExampleWithCustomName = Template.bind({});
ExampleWithCustomName.args = { name: 'SomeOtherName' };
