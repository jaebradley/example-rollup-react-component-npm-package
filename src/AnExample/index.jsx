import React from 'react';
import PropTypes from 'prop-types';

import './AnExample.css';

const AnExample = ({ name }) => (
  <div className="an-example">
    This is an example by
    { name }
  </div>
);

AnExample.propTypes = {
  name: PropTypes.string,
};

AnExample.defaultProps = {
  name: 'jaebaebae',
};

export default AnExample;
