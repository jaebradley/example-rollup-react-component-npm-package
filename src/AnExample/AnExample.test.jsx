import React from 'react';
import {
  shallow,
} from 'enzyme';
import renderer from 'react-test-renderer';

import AnExample from '.';

describe('AnExample', () => {
  describe('when name prop is defined', () => {
    it('renders name prop', () => {
      const anExample = shallow(<AnExample name="Some Name" />);
      expect(anExample.text()).toEqual('This is an example bySome Name');
    });

    it('renders correctly', () => {
      const tree = renderer.create(<AnExample name="Some Name" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when name prop is not defined', () => {
    it('renders with default name prop', () => {
      const anExample = shallow(<AnExample />);
      expect(anExample.text()).toEqual('This is an example byjaebaebae');
    });

    it('renders correctly', () => {
      const tree = renderer.create(<AnExample />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
