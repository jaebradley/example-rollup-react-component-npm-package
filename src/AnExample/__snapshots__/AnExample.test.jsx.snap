// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnExample when name prop is defined renders correctly 1`] = `
<div
  className="an-example"
>
  This is an example by
  Some Name
</div>
`;

exports[`AnExample when name prop is not defined renders correctly 1`] = `
<div
  className="an-example"
>
  This is an example by
  jaebaebae
</div>
`;
